week2assignment1.java

```diff
@@ -0,0 +1,111 @@
+/*
+ *Task 1: Valid Parentheses Combination Generator
+
+ *Task 2: Digit Sum Loop(String)
+               
+ *Task 3: Consecutive Number Summer
+            
+*Task 4: Caesar Cipher with Shift Variability
+           
+*Task 5: Encoding Challenge with ASCII Conversion
+           
+*Owner-->Pratul Agarwal
+*DATE-->09/09/2024
+ */
+
+package assignment_java_programming;
+import java.util.Scanner;
+
+
+public class week2assignment1 {
+    public static void main(String[] args) {
+        Scanner Inputfromuser = new Scanner(System.in);
+        String input;
+
+        do {
+            System.out.println(Constants.INDEX);
+            System.out.println(Constants.TASKNUMBER);
+            input = Inputfromuser.nextLine();
+            if (input.equalsIgnoreCase("exit")) {
+                System.out.println(Constants.EXIT);
+            } else {
+                int taskNumber;
+                try {
+                    taskNumber = Integer.parseInt(input);
+                } catch (NumberFormatException e) {
+                    System.out.println(Errors.INPUTERROR);
+                    continue;
+                }
+                switch (taskNumber) {
+                    case 1:
+                        validParenthesesCombinationGenerator(Inputfromuser);
+                        break;
+                    case 2:
+                        digitSumLoop(Inputfromuser);
+                        break;
+                    case 3:
+                        //consecutiveNumberSummer();
+                        break;
+                    case 4:
+                        //caesarCipherWithShiftVariability(Inputfromuser);
+                        break;
+                    case 5:
+                     //   encodingChallenge(Inputfromuser);
+                        break;
+                    default:
+                        System.out.println(Errors.INVALIDTASKNUMBER);
+                        break;
+                }
+            }
+        } while (!input.equalsIgnoreCase("exit"));
+
+        Inputfromuser.close();
+    }
+
+    private static void validParenthesesCombinationGenerator(Scanner Inputstring) {

```

Comment:
describe each method in brief through comments including short definition , input parameters and return values .